spec.bs

```diff
@@ -491,19 +510,83 @@ spec: html; urlPrefix: https://www.rfc-editor.org/rfc/
+      1. Let |topicMaxLength| be 3 (i.e. [=browsing topics types/topic id=] has maximum 3 digits).
+      1. Let |versionMaxLength| be 13 (i.e. chrome.1:1:11).
+      1. Let |paramsSeparatorLength| be 2 (i.e. structured fields use two characters (';' and '=') for each parameter)
+      1. Let |paramKeyLength| be 1 (i.e. always use a single character parameter key)
+      1. Let |innerListItemsSeparatorLength| be 1 (i.e. structured fields use a single character (' ') to separate inner list items).
+      1. Let |maxPaddingLength| be <code>|maxNumberOfEpochs| * |topicMaxLength| + |numVersionsInEpochs| * (|versionMaxLength| + |paramsSeparatorLength| + |paramKeyLength|) + (|maxNumberOfEpochs| - 1) * |innerListItemsSeparatorLength|</code>.
+
+      Note: the numbers used in the step above are specific to one of Chrome's implementation. Implementers should use numbers compatible with their configuration.
+
+      1. Let |paddingLength| be the maximum of |maxPaddingLength| - |serializedTopicsInnerListLength| and 0.
+      1. Let |paddedToken| be "P".
+      1. Append |paddingLength| <code>"0"</code> characters to the end of |paddedToken|.
+      1. Let |headerStructuredFields| be an empty Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-dictionaries">Dictionary</a>.
+      1. Set |headerStructuredFields|["<code>t</code>"] to |topicsInnerList|.
+      1. Set |headerStructuredFields|["<code>p</code>"] to a Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-tokens">Token</a> having words |paddedToken|.
       1. [=Set a structured field value=] given ([:Sec-Browsing-Topics:], |headerStructuredFields|) in |request|'s [=request/header list=].
 
-    Note: In Chrome's experimentation phase, it will additionally require a valid <a href="https://github.com/GoogleChrome/OriginTrials/blob/gh-pages/explainer.md">Origin Trial</a> token to exist in |initiatorWindow|'s <a data-cite="!HTML#concept-document-window">associated document</a> for the request to be eligible for topics.
+    <div class="note">
+      This algorithm transforms the topics list into structured fields format, which contains paddings to make the total length consistent for different topics callers.
+      <div class="example">
+          Empty returned topics, and underlying epochs have same versions:
+
+          <code>t=(), p=P000000000000000000000000000</code>
+      </div>
+      <div class="example">
+          One returned topic, and underlying epochs have same versions:
+
+          <code>t=(1;v=chrome.1:1:2), p=P00000000000</code>
+      </div>
+      <div class="example">
+          Three returned topics, and underlying epochs have same versions:
+
+          <code>t=(100;v=chrome.1:1:20 200 300), p=P</code>
+      </div>
+      <div class="example">
+          One returned topics, and underlying epochs have two different versions:
+
+          <code>t=(1;v=chrome.1:1:2 2), p=P0000000000000000000000000</code>
```

Comment:
I only see one version here?

Reply:
Done. (Removed. As example are less applicable given we'd always expose all versions.)